### PR TITLE
[receiver/azuremonitor] reduce public api

### LINKED ISSUE #24850: [receiver/azuremonitorreceiver] Consider reducing Go API
### Component(s)

receiver/azuremonitor

### Describe the issue you're reporting

The current Go API exposes the following items:
```
{
  "structs": [
    "ArmClient",
    "Config",
    "MetricsDefinitionsClientInterface",
    "MetricsValuesClient"
  ],
  "functions": [
    {
      "name": "NewFactory",
      "receiver": "",
      "return_types": [
        "receiver.Factory"
      ]
    }
  ]
}
```

Are any of those elements a good target to be unexported?

### PR DESCRIPTION
**Pull request by nslaughter**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>
resolves #24850

**Testing:** <Describe what testing was performed and which tests were added.>
Does not add tests.

**Documentation:** <Describe the documentation added.>
Does not change documentation.

### CODE REVIEW

**Comment by nslaughter** [comment]
I think this qualifies for no-changelog as it's just making some types private. However, I'm also open to making an entry if it's preferred.

**Comment by evan-bradley** [comment]
@nslaughter Could you add a changelog entry with an entry type of `api` that calls out the removed types? It's highly unlikely anyone is using the API, but technically this is a breaking change for the module.

File: receiver/azuremonitorreceiver/scraper.go [hunk 1]
```diff
@@ -134,27 +134,27 @@ func (s *azureScraper) getArmClientOptions() *arm.ClientOptions {
 	return &options
 }
 
-func (s *azureScraper) getArmClient() ArmClient {
+func (s *azureScraper) getArmClient() armClient {
```
Comment (michalpristas) [inline receiver/azuremonitorreceiver/scraper.go:137]: while it's ok to keep `armClient` and  `metricsDefinitionsClientInterface` interfaces to help mock objects in tests it's more idiomatic to return concrete types. 
also not related to this change, but this func is used only in `start` which has proper error handling (returning error) but we are silently failing here. this is just delaying problem to few seconds later risking panics when client is nil

Comment (nslaughter) [reply receiver/azuremonitorreceiver/scraper.go:137]: Appreciate you raising these points @michalpristas as this improves correctness and maintainability. To keep this PR focused I'd like to open issues based on your points and assign myself to fix in separate PRs.

**Review by codeboten — APPROVED** [review]

**Pull request merged**